### PR TITLE
Copter: re-order methods in Copter.h

### PR DESCRIPTION
**Pull request by rmackay9**
This is a non-functional change to re-order the methods listed in Copter.h so they are grouped by file and order within the file.

This will likely cause some merge conflicts for people but I think/hope that longer term it will make the code a little bit easier to understand.

Doing this organisation also highlighted some functions that were defined but never called nor implemented.  Also when looking through the list it makes me think we have some methods in the wrong .cpp files.

### CODE REVIEW

File: ArduCopter/Copter.h [hunk 1]
```diff
@@ -849,83 +904,75 @@ class Copter : public AP_HAL::HAL::Callbacks {
...
+    // UserCode.cpp
+    void userhook_init();
+    void userhook_FastLoop();
+    void userhook_50Hz();
+    void userhook_MediumLoop();
+    void userhook_SlowLoop();
+    void userhook_SuperSlowLoop();
 
-    Copter::Mode *flightmode;
+#include "mode.h"
```
Comment (OXINARF) [inline ArduCopter/Copter.h]: This is the only thing I don't like. Includes in the middle of a file is an ugly thing.

**Comment by Pedals2Paddles** [comment]
I was about to start cursing to myself since new PRs and commits like this keep blowing up my landing gear PR with merge conflicts. Then I looked at the change.  **Bless your heart, this is beautiful**.

File: ArduCopter/Copter.h (hunk 1 above)
Comment (rmackay9) [reply ArduCopter/Copter.h]: Indeed, i wanted to move it up but I don't think we can.  This is an existing issue in master of course.

**Comment by rmackay9** [comment]
OK, glad you guys like it.  merged!

**Pull request merged**